Minor dev experience improvement for Linux/Mac devs

jabba2324 opened the pull request:
Some Linux distros (namely ubuntu) now only ship with python 3 using the command 'python3' this will cause your launch-game.sh script to fail at detecting the game engine path.

The following changes add a retry for python3 and some logging around when the path is empty.

abcdefg30 commented on OpenRA.Game/Game.cs:
```diff
@@ -360,6 +360,11 @@ static void Initialize(Arguments args)
 
+				if(String.IsNullOrEmpty(launchPath))
+				{
+					Console.WriteLine("No launch path was specified, please check launch scripts.");
```
Our windows scripts don't set a launch path at all, so this warning would occur there all the time.

jabba2324 replied:
Do you know what the default value of launchPath is on Windows? If it was null or empty then the proceeding line 

`if (launchPath.First() == '"' && launchPath.Last() == '"')` 

Would always throw an exception as it currently does in Debian/Ubuntu? 

EDIT:
After a peek into the arguments class it seems that this would work fine on Windows as the default is 
`Assembly.GetEntryAssembly().Location `
which is implemented on Windows.

pchote replied:
My preference would be to not change this code at all - the issue is in the launch scripts.

abcdefg30 commented on launch-game.sh:
```diff
@@ -1,6 +1,12 @@
 
+if [ -z "$MODLAUNCHER" ]
+then
+      echo "\$MODLAUNCHER is empty trying python3"
```
Style nit: This should use tabs instead of spaces.

abcdefg30 reviewed (commented):
I wonder if we should use python3 by default here, and either just remove the normal python way or use that as fallback.

pchote commented:
Yes, python 2 is now obsolete so should not be treated as the go-to option.
I would prefer it if we could use our standard pattern for fallback commands:
```bash
if command -v python3 >/dev/null 2>&1; then
	 MODLAUNCHER=$(python3 -c "import os; print(os.path.realpath('$0'))")
else
	 MODLAUNCHER=$(python2 -c "import os; print(os.path.realpath('$0'))")
fi
```

It is more verbose, but makes the behaviour clear and avoids unnecessary warnings.

Mailaender commented:
I can confirm that this Debian specific workaround will have no bad influence on @openSUSE. Both `python` and `python3` point to Python 3 by default.

jabba2324 commented:
@pchote I'll implement the approach given in your comment and remove the logging from Game.cs

jabba2324 commented:
I haven't forgotten about this, just waiting for a new PC to arrive as my old one bit the dust.

pchote commented:
Can you please squash the commits together?

Commit:
Add logging for when launch path is not specified


add retry for python3 only distros


Revert "Add logging for when launch path is not specified"

This reverts commit 523538293c33ea02e3dc2009adb361178b9d3333.

use python3 by default

jabba2324 commented:
> Can you please squash the commits together?

Squashed

pchote commented:
Sorry for the delay here. The commit message no longer reflects the contents of what it actually does, so can we reword this to something like "Add python3 support to launch-game.sh"?

This otherwise LGTM - tested on Ubuntu 20.04 and macOS 10.15.

jabba2324 commented:
> Sorry for the delay here. The commit message no longer reflects the contents of what it actually does, so can we reword this to something like "Add python3 support to launch-game.sh"?
> 
> This otherwise LGTM - tested on Ubuntu 20.04 and macOS 10.15.

I've opened a new PR here with an appropriately named branch and commit -m

https://github.com/OpenRA/OpenRA/pull/18375
